src/html-conventions-lib/general-connectors-html-convention.xsl

```diff
@@ -688,4 +688,133 @@
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:choose>
+                <xsl:when test="$isDependency = fn:true()">
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Association']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is used in other connectors.'))
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Dependency']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is used in other connectors.'))
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:otherwise>
+            </xsl:choose>
+        </xsl:if>
+
+        <xsl:if test="boolean($targetRole)">
+            <xsl:choose>
+                <xsl:when test="$isDependency = fn:true()">
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($targetRole, root($connector))/properties[@ea_type = 'Association']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is used in other connectors.'))
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($targetRole, root($connector))/properties[@ea_type = 'Dependency']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is used in other connectors.'))
```

Comment:
Should we not tell where else is the connector name used (Source -> Target Classes)?

Reply:
Not sure. TBD